include/onnxruntime/core/framework/kernel_registry.h

```diff
@@ -44,6 +44,25 @@ class KernelRegistry {
                        const TypeConstraintMap& type_constraints,
                        const KernelCreateInfo** out) const;
 
+  /**
+   * @brief Find out whether a kernel is registered, without a node.
+   *        This should be useful in graph optimizers, to check whether
```

Comment:
>graph optimizers [](http://example.com/codeflow?start=37&length=16)

It looks like a tedious work to get type_constraints for insert_cast_transformer. If it is same case for other optimizer, it discourages users to use it?

Reply:
it would be much simpler for other optimizers. Another PR showing how it is usually used, soon.

Reply:
https://github.com/microsoft/onnxruntime/pull/15724/files#diff-930bf0c155479c9a2e17b4996b0ae133e7db12e3161b6ac868e6e9756d76771b

Here is what I hope it would be used. Usually the query code should be very similar to kernel definition code.